core/backend/src/IModelDb.ts

```diff
@@ -844,13 +861,31 @@ export abstract class IModelDb extends IModel {
    * @alpha
    */
   public async importSchemaStrings(serializedXmlSchemas: string[]): Promise<void> {
-    if (this.iTwinId && this.iTwinId !== Guid.empty) // if this iModel is associated with an iTwin, importing schema requires the schema lock
-      await this.acquireSchemaLock();
+    if (serializedXmlSchemas.length === 0)
+      return;
 
-    const stat = this.nativeDb.importXmlSchemas(serializedXmlSchemas, { schemaLockHeld: true });
-    if (DbResult.BE_SQLITE_OK !== stat)
-      throw new IModelError(stat, "Error importing schema");
+    if (this.nativeDb.schemaSyncEnabled()) {
+      await SchemaSync.withLockedAccess(this, { openMode: OpenMode.Readonly, operationName: "schemaSync" }, async (syncAccess) => {
+        const schemaSyncDbUri = syncAccess.getUri();
+        this.saveChanges();
+        let stat = this.nativeDb.importXmlSchemas(serializedXmlSchemas, { schemaLockHeld: false, schemaSyncDbUri });
+        if (DbResult.BE_SQLITE_ERROR_SchemaLockFailed === stat) {
+          this.abandonChanges();
+          if (this.nativeDb.getITwinId() !== Guid.empty)
```

Comment:
I don't understand why you attempt to import again with the schema lock. If the iModel uses schemaSync, it should not attempt to bypass it.

Reply:
The schemaSync in most common cases does not require schema lock (_additive changes to the schema_). But in the rare case where schema import causes "data transform" it requires a schema lock just to ensure that it can perform data transform safely. It does not bypass schemaSync in the second attempt. We can rename SchemaLock  to iModelLock.

Reply:
Are you saying that the only way for `importXmlSchemas` to fail with `SchemaLockFailed` is that the import requires data transforms? If so, please put a comment to that effect here (because that's certainly not obvious) and explain why you are going to retry the import. 

SchemaLock is a fine name for the lock, and it's documented that it can't be acquired unless no other locks are held by any Briefcase. It is the equivalent of a "Db upgrade" requiring everyone to push all their changes before it can succeed. Hopefully that doesn't happen often.

Reply:
also, I assume that when you call `importXmlSchemas` with `schemaLockHeld` that it updates the SchemaSyncDb if it exists? Please document that here too.